xrpl/wallet/main.py

```diff
@@ -15,48 +13,43 @@ class Wallet:
     details.
     """
 
-    def __init__(self: Wallet, seed: str) -> None:
+    def __init__(self: Wallet, public_key: str, private_key: str) -> None:
         """
         Generate a new Wallet.
 
         Args:
-            seed: The seed from which the public and private keys are derived.
+            public_key: The public key for this Wallet.
+            private_key: The private key for this Wallet.
         """
-        #: The core value that is used to derive all other information about
-        #: this wallet. MUST be kept secret!
-        self.seed = seed
-
-        pk, sk = derive_keypair(self.seed)
         #: The public key that is used to identify this wallet's signatures, as
         #: a hexadecimal string.
-        self.pub_key = pk
+        self.public_key = public_key
```

Comment:
Public key should be derivable from the private key anyway

Reply:
I still think this should derive the public key from the private key rather than requiring the user to pass in both.